src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,72 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.tasks.Task;
+
+/**
+ * Represents a task list to store user's tasks.
+ */
+public class TaskList {
+    private List<Task> taskList;
+
+    /**
+     * Creates a task list object with ArrayList as the underlying data structure.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Adds a new task into the task list.
+     *
+     * @param newTask New Task to be added.
+     */
+    public void add(Task newTask) {
+        this.taskList.add(newTask);
+    }
+
+    /**
+     * Gets a task from the task list at a given index.
+     *
+     * @param index Index to retrieve Task from the task list.
+     * @return Task represented at given index.
+     */
+    public Task get(int index) {
+        return this.taskList.get(index);
+    }
+
+    /**
+     * Gets total number of tasks currently stored in the task list.
+     *
+     * @return Size of the task list.
+     */
+    public int getNumTask() {
+        return this.taskList.size();
+    }
+
+    /**
+     * Lists tasks currently in the task list.
+     *
+     * @return String representation of items in task list.
+     */
+    public String listItems() {
+        String listOfItemsInString = "";
```

Comment:
I really like the way you write your code, this is just a minor point that I would consider as nitpicking because your naming is really too neat. 

Could consider renaming listOfItemsInString as listOfItems**As**String or instead because it's the items within the list as a string instead of being within the string.